dffml/model/model.py

```diff
@@ -80,28 +88,87 @@ def __init__(self, config):
         # TODO Just in case its a string. We should make it so that on
         # instantiation of an @config we convert properties to their correct
         # types.
-        directory = getattr(self.config, "directory", None)
-        if isinstance(directory, str):
-            directory = pathlib.Path(directory)
-        if isinstance(directory, pathlib.Path):
-            # to treat "~" as the the home directory rather than a literal
-            directory = directory.expanduser().resolve()
-            self.config.directory = directory
+        location = getattr(self.config, "location", None)
+        if isinstance(location, str):
+            location = pathlib.Path(location)
+        if isinstance(location, pathlib.Path):
+            # to treat "~" as the the home location rather than a literal
+            location = location.expanduser().resolve()
+            self.config.location = location
+        self.location_is_file = self.config.location.is_file()
 
     def __call__(self) -> ModelContext:
-        self._make_config_directory()
+        self._make_config_location()
         return self.CONTEXT(self)
 
-    def _make_config_directory(self):
+    async def __aenter__(self):
+        if self.location_is_file:
+            location = self.config.location
+            temp_dir = self._get_directory()
+            if zipfile.is_zipfile(location):
+                await self._run_operation("extract", location, temp_dir)
+            self.config.location = temp_dir
+            self.config.file_location = location
+
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        if self.location_is_file:
+            config_path = self.config.location / "config.json"
+            config_path.write_text(json.dumps(self.config))
+            output_location = self.config.file_location
+            await self._run_operation(
+                "archive", self.temp_dir, output_location
+            )
+            shutil.rmtree(self.temp_dir)
+
+    async def _run_operation(self, action, input_path, output_path):
```

Comment:
- [ ] Let's do the rename in a separate PR than the implementation of the dataflow stuff
  - [ ] Looking at this, It seems like the best way to make progress is to wait to do the rename of directory to location until we have the dataflow stuff implemented. Since we can implement using `self.directory` and then do the rename later.

Reply:
So shall make this PR only for implementation of the dataflow stuff ? and yes the rename can be done after that.....

Reply:
ok sounds perfect to me :)